src/main/java/solutions/fairdata/openrefine/metadata/fdp/FairDataPointClient.java

```diff
@@ -0,0 +1,502 @@
+                    SimpleValueFactory.getInstance().createIRI(actualURI)
+            );
+            return DatasetTransformerUtils.metadata2DTO(returnDatasetMetadata);
+        } else {
+            throw new FairDataPointException(conn.getResponseCode(), conn.getResponseMessage());
+        }
+    }
+
+    /**
+     * Create a new distribution in FDP
+     *
+     * @param distributionDTO distribution to be created
+     * @return created distribution
+     * @throws IOException in case of a communication error
+     * @throws FairDataPointException in case that FDP responds with an unexpected code
+     */
+    public DistributionDTO postDistribution(DistributionDTO distributionDTO) throws IOException, MetadataException, FairDataPointException {
+        HttpURLConnection conn = createConnection(fdpBaseURI + DISTRIBUTION_PART, "POST", "text/turtle");
+        conn.addRequestProperty("Content-Type", "text/turtle");
+        conn.setDoOutput(true);
+
+        // Generate random IRI, FDP will replace it with really unique
+        String uri = fdpBaseURI + "/fdp/distribution/" + UUID.randomUUID();
+        distributionDTO.setIri(uri);
+
+        DistributionMetadata distributionMetadata = DistributionTransformerUtils.dto2Metadata(distributionDTO);
+        distributionMetadata.setIdentifier(MetadataTransformerUtils.createIdentifier(uri));
+        MetadataTransformerUtils.setTimestamps(distributionMetadata);
+        String metadata = MetadataUtils.getString(distributionMetadata, RDFFormat.TURTLE);
+
+        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(conn.getOutputStream(), StandardCharsets.UTF_8));
+        bw.write(metadata);
+        bw.flush();
+        bw.close();
+
+        if(conn.getResponseCode() == HttpURLConnection.HTTP_CREATED) {
+            String actualURI = conn.getHeaderField("Location");
+            DistributionMetadataParser distributionMetadataParser = new DistributionMetadataParser();
+
+            DistributionMetadata returnDistributionMetadata = distributionMetadataParser.parse(
+                    parseStatements(conn, actualURI),
+                    SimpleValueFactory.getInstance().createIRI(actualURI)
+            );
+            return DistributionTransformerUtils.metadata2DTO(returnDistributionMetadata);
+        } else {
+            throw new FairDataPointException(conn.getResponseCode(), conn.getResponseMessage());
+        }
+    }
+
+    /**
+     * Helper for parsing RDF statements from connection
+     *
+     * @param conn connection with FAIR Data Point
+     * @param uri base URI to be retrieved
+     * @return list of statements for given URI
+     * @throws IOException in case of communication error with FDP
+     */
+    private ArrayList<Statement> parseStatements(HttpURLConnection conn, String uri) throws IOException {
+        TurtleParser parser = new TurtleParser();
+        StatementCollector collector = new StatementCollector();
+        parser.setRDFHandler(collector);
```

Comment:
Could be replaced with the `RDFParser parser = Rio.createParser(RDFFormat.TURTLE);`, but the end result would be exactly the same. 🤷‍♂